### ISSUE TITLE
Model not in eval mode for baselines

### ISSUE THREAD

**Issue by josephenguehard** [JonathanCrabbe/Dynamask#8]
Hi,

There seems to be an error when computing the baselines for the state and mimic experiments: the model is not set in `eval` mode, which affects the results (probably due to the 0.5 dropout). For instance, on the state experiment using `deep_lift`, AUP improves from around 0.57 to around 0.94, and AUR from 0.17 to 0.35.

The line `self.base_model.eval()` should be added in the file `fit.TSX.explainers` for all the baselines. For instance:

``` python
class DeepLiftExplainer:
    def __init__(self, model, activation=torch.nn.Softmax(-1)):
        self.device = "cuda" if torch.cuda.is_available() else "cpu"
        self.base_model = model.to(self.device)
        self.explainer = DeepLift(self.base_model)
        self.activation = activation

    def attribute(self, x, y, retrospective=False):
        self.base_model.zero_grad()
        self.base_model.eval()  # Missing line
        if retrospective:
            score = self.explainer.attribute(x, target=y.long(), baselines=(x * 0))
            score = abs(score.detach().cpu().numpy())
        else:
            score = np.zeros(x.shape)
    ...
```
